packages/backend-app-api/src/wiring/types.ts

```diff
@@ -39,6 +39,7 @@ export interface CreateSpecializedBackendOptions {
 
 export interface ServiceHolder {
   get<T>(api: ServiceRef<T>, pluginId: string): Promise<T> | undefined;
+  checkForCircularDeps(): void;
```

Comment:
Hmm, this doesn't quite make sense to add to any of these interfaces. Given that, I'm feeling it makes the most sense to keep `checkForCircularDeps` private and instead add a static `ServiceRegistry.create()` factory that instantiates the registry and checks for circular deps, removing the call from `BackendInitialized.start`

Reply:
Updated to have a static create method and private constructor. I've made the changeset a minor bump (IIRC that's fine for breaking changes pre v1.0.0)